mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+    a list using the interactions argument::
+
+        ens = Ensemble(dirname='molecule', interactions=['Coulomb']
+
+    .. versionadded:: 0.8.0
+    """
+
+    def __init__(self, dirname=None, solvents=('octanol', 'water'),
+                 topology_paths=None, interactions=('Coulomb', 'VDW')):
+        self.top_dict = topology_paths
+        self._num_systems = 0
+        self._ensemble = {}
+        self._keys = []
+        self._solvents = solvents
+        self._interactions = interactions
+
+        if dirname is None:
+            return
+
+        if not os.path.exists(dirname):
+            logger.error(f"Directory {dirname} does not exist")
+            raise FileNotFoundError
+
+        self._ensemble_dir = dirname
+        self._build_ensemble()
+
+    def __repr__(self):
+        return f"<Ensemble Containing {self._num_systems} System>"
+
+    def __len__(self):
+        return self._num_systems
+
+    def __getitem__(self, index):
+        """Allows dictionary like indexing"""
+        return self._ensemble[index]
+
+    @staticmethod
+    def _sort_trajectories(trajectories: list):
+        """sort list of trajectory files alphabetically"""
+        return sorted(trajectories)
+
+    @staticmethod
+    def _sort_topologies(topologies: list):
+        """sorts list of trajectory files with .tpr first"""
+        top = []
+        logger.info('If more than one topology is present the tpr will be the one used')
+        for i in range(len(topologies)):
+            file = topologies[i]
+            if file.endswith('.tpr'):
+                topologies.pop(i)
+                top = [file] + topologies
+                break
+        return top
+
+    def _load_universe_from_dir(self, solvent):
+        """Loads system in directory into an MDAnalysis Universe
+
+        logs warning if more than one topology is in directory. If
+        more than one trajectory attempts to load both of them
+        in a universe if that fail will try to load each individually"""
+        cur_dir = os.listdir(os.curdir)
+        self.trj = []
+        self.top = []
+
+        if not self.top_dict is None:
+            # if top is specified in kwargs, saved to list
+            self.top = [self.top_dict[solvent]]
+
+        for file in cur_dir:
+            if file.endswith('.xtc'):
+                # Saving trajectory directories
+                self.trj.append(file)
+            elif (file.endswith('gro') or file.endswith('.tpr') or file.endswith('gro.b2z') or file.endswith('gro.gz'))\
+                    and self.top_dict is None:
+                # Saving topology directories
+                self.top.append(file)
+
+        if len(self.top) == 0 or len(self.trj) == 0:
+            logger.warning('No MD files detected in %s', os.curdir)
+            raise NoDataWarning
+
+        self.trj = self._sort_trajectories(self.trj)
+        if len(self.top) > 1:
+            self.top = self._sort_topologies(self.top)
```

Comment:
Now 
1. make the trajectory paths absolute
2. load the universe

If you have to protect the Universe creation with try/except, please also display the original exception so that one has a chance to debug things. In fact, it's probably best to just not do anything here and let any exceptions just happen. Start with 

```python
trajectory = [os.path.abspath(trj) for trj in util.asiterable(trajectory)]   # asiterable() _may_ be in util... or MDA?
return mda.Universe(topology, trajectory, **universe_kwargs)
```

Pass `universe_kwargs` through so that you have a fighting chance to modify universe creation.

Reply:
done, added the path addition to `sort_topologies` and pass the kwargs